packages/vitest/src/node/reporters/github-actions.ts

```diff
@@ -0,0 +1,96 @@
+import { Console } from 'node:console'
+import { Writable } from 'node:stream'
+import { getTasks } from '@vitest/runner/utils'
+import stripAnsi from 'strip-ansi'
+import type { File, Reporter, Vitest } from '../../types'
+import { getFullName } from '../../utils'
+import { printError } from '../error'
+import { Logger } from '../logger'
+import type { WorkspaceProject } from '../workspace'
+
+export class GithubActionsReporter implements Reporter {
+  ctx: Vitest = undefined!
+
+  onInit(ctx: Vitest) {
+    this.ctx = ctx
+  }
+
+  async onFinished(files: File[] = []) {
```

Comment:
I forgot "unhandled errors" but probably it's possible to handle as well (though it can be only associated with "core project" since there's no file association).